oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java

```diff
@@ -666,19 +730,28 @@ static ReadBatchResponse readFrom(ByteBuf buffer) {
             }
 
             int compositedElementCount = buffer.readInt();
-            ByteBuf compositedElementLengths =
-                    buffer.readRetainedSlice(Integer.BYTES * compositedElementCount);
-
+            boolean compressEnabled = buffer.readBoolean();
+            int contentLength = buffer.readInt();
+            ByteBuf compositedElementLengths = null;
             ByteBuf[] dataBuffers = new ByteBuf[columnCount];
-            for (int i = 0; i < columnCount; i++) {
-                dataBuffers[i] = buffer.readRetainedSlice(dataBufferLengths[i]);
-            }
-
             ByteBuf[] nullBuffers = new ByteBuf[columnCount];
-            for (int i = 0; i < columnCount; i++) {
-                nullBuffers[i] = buffer.readRetainedSlice(rowCount);
+            try {
+                ByteBuf content = compressEnabled ?
+                        ZStdUtils.decompress(buffer.retainedSlice(), contentLength)

```

Comment:
buffer created by `buffer.retainedSlice()` does not get released in after decompression.

Reply:
I did that in decompress method.

Reply:
ok. I didn't notice that